avro-serializer/src/main/java/io/confluent/kafka/formatter/AvroMessageFormatter.java

```diff
@@ -98,13 +107,29 @@ public void init(Properties props) {
     if (props.containsKey("line.separator")) {
       lineSeparator = props.getProperty("line.separator").getBytes();
     }
+    if (props.containsKey("key.deserializer")) {
+      try {
+        keyDeserializer =
+            (Deserializer)Class.forName((String) props.get("key.deserializer")).newInstance();
+      } catch (Exception e) {
+        throw new ConfigException("Error initializing Key deserializer", e);
+      }
+    }
   }
 
   @Override
   public void writeTo(ConsumerRecord<byte[], byte[]> consumerRecord, PrintStream output) {
     if (printKey) {
       try {
-        writeTo(consumerRecord.key(), output);
+        if (keyDeserializer != null) {
+          Object deserializedKey = consumerRecord.key() == null ? null :
```

Comment:
I like to lay ternary operator expressions out like:

```
consumerRecord.key() == null 
  ? null 
  : keyDeserializer.deserialize(null, consumerRecord.key())
```

but I don't know what our standard is and i'm fine to leave it like this